### PR TITLE
[CSFS-111] Develop a Customer Details view page

### PR DESCRIPTION
**Pull request by Tocyn**
Screenshot 2021-02-21 at 12 12 40

### CODE REVIEW

**Comment by Tocyn** [comment]
@tunderabiu2@gmail.com <tunderabiu2@gmail.com> do you want me to change the
error message

File: submissions/oluwatosinOke/phpAndDatabases/customerlist.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+require_once("cookie.php");
+try {
+  	$sql = "SELECT * FROM customers";
```
Comment (adaugochi) [inline submissions/oluwatosinOke/phpAndDatabases/customerlist.php:4]: What is on your `cookie.php` is the same thing you are doing here.

That aside, the idea of this task is to store the result returned from this SQL query in a cookie, Each time I reload this page, I don't want to always run this SQL query but instead, I want to get it from the data stored in the cookie as long as the cookie has not expired. If the cookie is has expired then run the SQL query

With your current implementation, each time I reload this page, I am always executing this SQL query.

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by Tocyn** [comment]
@adaa@cottacush.com <adaa@cottacush.com>
How can I sort it out

File: submissions/oluwatosinOke/phpAndDatabases/cookie.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    $sql = "SELECT * FROM customers";
+    $result= $pdo->query($sql);
+    // setting up the cookie to save the result gotten from the databse
+    setcookie("customers", "view", time()+7200, "/");
+    return $result;
+} catch (PDOException $e) {
+die ('haaa...could not execute ' . $sql . $e->getMessage());
+}
+
+$result = isset($_COOKIE['view']) ? unserialize($_COOKIE['view']) : $view->list($id);
```
Comment (adaugochi) [inline submissions/oluwatosinOke/phpAndDatabases/cookie.php:13]: Where is `$view->list($id)` coming from?

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by Tocyn** [comment]
@adaugochi 
Kindly Review.
Thanks.

File: submissions/oluwatosinOke/phpAndDatabases/customerlist.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
+require_once("cookie.php");
+    if ($result->rowCount()>0){
```
Comment (adaugochi) [inline submissions/oluwatosinOke/phpAndDatabases/customerlist.php:3]: This `$result` is what is always `return` each time you run the `$sql` query in the `cookie.php`. 

What we want is that the `$result` used here to either come from the `$sql`query (if cookie is not set) or from cookie (if cookie is set). You are already storing the data in a cookie called `view` ( see line 8 of cookie,php file) but that's all you are doing. You are not getting it or using it

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by Tocyn** [comment]
@adaugochi 
Kindly Review.

File: submissions/oluwatosinOke/phpAndDatabases/customerlist.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<?php
+require_once("cookie.php");
+$queryDb = new DbQuery();
+$results = isset($_COOKIE['view'])? unserialize($_COOKIE['view']) : $queryDb->getAllCustomers();
```
Comment (adaugochi) [inline submissions/oluwatosinOke/phpAndDatabases/customerlist.php:4]: Where is this `$queryDb->getAllCustomers()` coming from?

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by Tocyn** [comment]
@adaugochi 
Kindly Review.

File: submissions/oluwatosinOke/phpAndDatabases/connection.php [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?php
+//connecting to the mysql database
+  
+try {
+  $pdo = new PDO("mysql: host=localhost; dbname=e_store", "root", "tocyn");
```
Comment (adaugochi) [inline submissions/oluwatosinOke/phpAndDatabases/connection.php:5]: In your eCommerce task, this is not what you called your database name

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by Tocyn** [comment]
Changes have been effected.
Kindly review.

**Review by adaugochi — APPROVED** [review]

**Pull request merged**